jmix-core/core/src/main/java/io/jmix/core/FetchPlans.java

```diff
@@ -27,13 +26,17 @@
 public class FetchPlans {
 
     @Autowired
-    protected ObjectProvider<FetchPlanBuilder> fetchPlanBuilderProvider;
+    protected Metadata metadata;
+    @Autowired
+    protected MetadataTools metadataTools;
+    @Autowired
+    protected FetchPlanRepository fetchPlanRepository;
 
     /**
      * Returns {@link FetchPlan} builder for the given entity class.
      */
     public FetchPlanBuilder builder(Class<?> entityClass) {
-        return fetchPlanBuilderProvider.getObject(entityClass);
+        return new FetchPlanBuilder(this, entityClass);
```

Comment:
I would prefer passing all dependencies separately. Is it possible?

Reply:
Sure, I updatred PR.